### PR TITLE
[Filter] Remove gst dep from single-shot

### PR DESCRIPTION
**Pull request by gichan-jang**
- Remove gst dependency from single-shot.
 - Separate single-shot package from nnstreamer(pipeline)
 - nnstreamer depends nnstreaner-single package. 
- Remove gst dependency from tensor filter subplugins.
  - tensorflow, tensorflow-lite, pytorch, nnfw,caffe2
 - Remove gst dependency from singleshot test

**Self evaluation:**
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ *]Passed [ ]Failed [ ]Skipped

Signed-off-by: gichan <gichan2.jang@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/include/nnstreamer_plugin_api_single.h [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+ */
+
+#define _STR_NULL(str) ((str) ? (str) : "(NULL)")
+
+/**
+ * @brief Initialize the tensor info structure
+ * @param info tensor info structure to be initialized
+ */
+extern void
+gst_tensor_info_init (GstTensorInfo * info);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/include/nnstreamer_plugin_api_single.h:34]: Discussion later : let's consider not to add this header.
IMO, util functions in nnstreamer_plugin_api.h / other gst-depended function (e.g. caps-from-pad) can be moved into tensor_common.h.

File: packaging/nnstreamer.spec [hunk 2]
```diff
@@ -479,6 +485,7 @@ NNStreamer's tensor_filter subplugin of trix-engine
 %package devel
 Summary:	Development package for custom tensor operator developers (tensor_filter/custom)
 Requires:	nnstreamer = %{version}-%{release}
+Requires:	nnstreamer-single = %{version}-%{release}
```
Comment (myungjoo) [inline packaging/nnstreamer.spec:488]: ```s/nnstreamer-single/nnstreamer-single-devel/```

**Comment by myungjoo** [comment]
[  271s] Processing files: nnstreamer-single-2.1.0-0.aarch64
[  271s] Aptk: PG1hbmlmZXN0PgogPHJlcXVlc3Q+CiAgICA8ZG9tYWluIG5hbWU9Il8iLz4KIDwv
[  271s] cmVxdWVzdD4KPC9tYW5pZmVzdD4K
[  271s] 
[  271s] Executing(%license): /bin/sh -e /var/tmp/rpm-tmp.91wxpU
[  271s] + umask 022
[  271s] + cd /home/abuild/rpmbuild/BUILD
[  271s] + cd nnstreamer-2.1.0
[  271s] + LICENSEDIR=/home/abuild/rpmbuild/BUILDROOT/nnstreamer-2.1.0-0.aarch64/usr/share/licenses/nnstreamer-single
[  271s] + export LC_ALL=C
[  271s] + LC_ALL=C
[  271s] + export LICENSEDIR
[  271s] + /bin/mkdir -p /home/abuild/rpmbuild/BUILDROOT/nnstreamer-2.1.0-0.aarch64/usr/share/licenses/nnstreamer-single
[  271s] + cp -pr LICENSE /home/abuild/rpmbuild/BUILDROOT/nnstreamer-2.1.0-0.aarch64/usr/share/licenses/nnstreamer-single
[  271s] + exit 0
[  271s] Provides: libnnstreamer-single.so()(64bit) nnstreamer-single = 2.1.0-0 nnstreamer-single(aarch-64) = 2.1.0-0
[  271s] Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
[  271s] Requires: ld-linux-aarch64.so.1()(64bit) ld-linux-aarch64.so.1(GLIBC_2.17)(64bit) libc.so.6()(64bit) libc.so.6(GLIBC_2.17)(64bit) libdlog.so.0()(64bit) libgcc_s.so.1()(64bit) libgcc_s.so.1(GCC_3.0)(64bit) libgcc_s.so.1(GCC_3.3.1)(64bit) libglib-2.0.so.0()(64bit) libgmodule-2.0.so.0()(64bit) libgobject-2.0.so.0()(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.17)(64bit)
```

File: gst/nnstreamer/include/nnstreamer_plugin_api_single.h (hunk 1 above)
Comment (gichan-jang) [reply gst/nnstreamer/include/nnstreamer_plugin_api_single.h]: @jaeyun-jung , I changed header file name and remove some functions. Please check it again. (https://github.com/nnstreamer/nnstreamer/pull/3652/commits/a6f6077eea9e302e0b7b2f29d85d11b74ea94989)

File: packaging/nnstreamer.spec (hunk 2 above)
Comment (gichan-jang) [reply packaging/nnstreamer.spec]: Thanks. Updated.

File: gst/nnstreamer/nnstreamer_internal.h [hunk 3]
```diff
@@ -4,7 +4,7 @@
  *
  * @file	nnstreamer_internal.h
  * @date	28 Jan 2021
- * @brief	Internal header for NNStreamer plugins and native APIs.
+ * @brief	Internal header for NNStreamer plugins and native sintgle-shot APIs.
```
Comment (niley7464) [inline gst/nnstreamer/nnstreamer_internal.h:7]: Maybe sintgle is a typo for single..?

**Review by myungjoo — APPROVED** [review]
LGTM

File: gst/nnstreamer/nnstreamer_internal.h (hunk 3 above)
Comment (gichan-jang) [reply gst/nnstreamer/nnstreamer_internal.h]: Thanks. Updated.

**Review by niley7464 — APPROVED** [review]
Looks fine to me :)

**Pull request merged**